drivers/serial/uart_ns16550.c

```diff
@@ -403,6 +407,10 @@ static int uart_ns16550_configure(const struct device *dev,
 	}
 #endif
 
+#ifdef CONFIG_SOC_AM62x_M4
```

Comment:
Could this be something more generic like `CONFIG_UART_NS16550_16X_MODE`?
And rename `MDR1_UART_MODE` -> `MDR1_UART_16X_MODE`

Reply:
What is 16X mode? I can't seem to find the config.

Reply:
I was suggesting to make a new config for the NS16550 UART.
This UART driver is used by many different SOCs.
Avoiding specific configs (`CONFIG_SOC_AM62x_M4`) in here will help keep this file tidy and reduce how frequently it is modified.

> What is 16X mode? 

I think that is the mode you are selecting in the MDR1 register -> 16X oversampling

Maybe one of the code owners could comment on how to best do this? @dcpleung @nashif @aasthagr

Reply:
I will open a new PR for this, and add a variant for TI K3, as there are more registers for modes.